File: ui/src/app/models/Flight.ts

export interface IFlight {
    id: string;
    code: string;
    arrival: string;
    depature: string;
    totalSeats: number;
    price: number;
}

export interface IFlightDeatils extends IFlight {
    flightSchedulers: IFlightScheduler[],
    seats: Seat[]

}

export interface IFlightScheduler {
    id: String;
    flightId: string;
    departureDateTime: Date;
    arrivalDateTime: Date;
    journeyTime: string;
}

export interface Seat {
    id: string;
    flightId: string;
    seatNumber: string
}

File: TravelAgentAPI/TravelAgentAPI/DataModels/Flight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TravelAgentAPI.DataModels
{
    public class Flight : Common
    {
        public string Code { get; set; }

        public string Arrival { get; set; }

        public string Depature { get; set; }

        public int TotalSeats { get; set; }
        public Decimal Price { get; set; }

        public virtual List<FlightScheduler> FlightSchedulers { get; set; }

        public virtual List<Seat> Seats { get; set; }

    }
}
